### PR TITLE
Externalize report assets and remove Git LFS for HTML exports

### PR DESCRIPTION
**Pull request by z0u**
## Description

This PR refactors how Marimo notebook exports are published, moving from Git LFS storage to a content-addressed asset store. The key changes:

1. **Remove Git LFS for HTML exports**: Updated `.gitattributes` to stop tracking `docs/**/__marimo__/**` files with Git LFS. Reports now commit as plain HTML files.

2. **Externalize heavy assets**: Added `use_publisher()` and `report_bundle()` integration to reports (`gpt-sweep/report.py`, `pipeline/report.py`, `themed.py`, `gpt.py`) so that figures and other assets are written to a durable store (HF bucket when configured) rather than embedded in HTML. This keeps exported HTML files lightweight.

3. **Update build pipeline**: Modified `scripts/build_site.py` to handle asset externalization with idempotent, read-only-safe publishing. Already-published assets are skipped to avoid redundant writes.

4. **Add resolver support for external source links**: Extended `test_build_site.py` and the resolver to handle links to source modules outside `docs/` that should resolve to GitHub URLs, ensuring they survive the asset `<base>` tag rewriting.

5. **Store API enhancement**: Added `is_published()` method to `HFStore` to check if an asset is already published.

6. **Remove obsolete notebook**: Deleted `docs/gpt_sweep.py` (the interactive sweep notebook), keeping only the report-based workflow.

7. **Update documentation**: Refreshed `docs/index.md` and apparatus references to reflect the new asset publishing model.

The net effect: reports are now lighter, assets are content-addressed and reusable, and the CI Pages build can efficiently republish without redundant uploads.

https://claude.ai/code/session_01Ux5HZjmcbhf1iScXbBFLjK

### CODE REVIEW

File: .agents/skills/mi-ni/references/apparatus.md [hunk 1]
```diff
@@ -39,6 +39,23 @@ await app.volume.download('outputs', f'/data/outputs')
...
+app = ModalApparatus('demo').w(gpu='L4') if app_type == 'modal' else LocalApparatus('demo')
+```
+
+Export it headless with `./go run` (it wraps `marimo export` and already supplies the `--` separator); pass notebook options **after** the notebook:
+
+```bash
+./go run docs/gpt.py --app=modal --arch=ngpt
+```
+
+**Syntax gotcha:** marimo's `cli_args()` only parses `--key=value` or `--key value`. A bare `key=value` (or a stray extra `--`) parses to *nothing*, so the notebook silently falls back to its default (here `local`) with no error. Confirm which backend actually ran from the logs — a Modal run prints `Creating Modal image …` then `Running … on Modal`; a local one prints `Running … locally on CPU`.
```
Comment (z0u) [inline .agents/skills/mi-ni/references/apparatus.md]: I think a delimiting `--` is actually ok.

```bash
./go run docs/foo/report.py -- --app=modal
```

is probably preferable.

File: .gitattributes [hunk 2]
```diff
@@ -1,2 +1,4 @@
-docs/**/large-assets/** filter=lfs diff=lfs merge=lfs -text
-docs/**/__marimo__/** filter=lfs diff=lfs merge=lfs -text
+# Reports (Marimo HTML exports) commit to Git as plain files: figures and other
+# heavy assets are externalized to the HF bucket at build time, so the committed
+# HTML is just a light shell. The externalized `_assets/` are gitignored (durable
```
Comment (z0u) [inline .gitattributes]: Actually I thought we were planning to keep the HTML in the HF bucket (or configured Store) too? IIRC the plan was to put both the HTML and assets in the bucket. Then in GH Actions, copy only the HTML to GH Pages and insert the `base` tag. However I'm a little unsure how the HTML was meant to get into the bucket initially when running a report.

File: docs/__marimo__/getting_started.html
Comment (z0u) [inline docs/__marimo__/getting_started.html:1]: As mentioned above: I'm not sure should be in git? If we can push it to the Store, it could be ignored instead.

File: docs/gpt-sweep/experiment.py [hunk 3]
```diff
@@ -123,9 +128,29 @@ def train_one(config, arch_label: str, lr_str: str) -> tuple:
...
+    bucket token), so the report can resolve the curves by name and render them —
+    keeping the data in the durable store rather than a file committed to Git.
+    Idempotent: ``put`` is content-addressed and ``set_ref`` is last-writer-wins.
+    """
+    import json
+
+    from mini.runs import data_root
+    from mini.store import default_store, store_root_for
+
+    store = default_store(store_root_for(data_root() / 'gpt-sweep'))
```
Comment (z0u) [inline docs/gpt-sweep/experiment.py]: Do we need to explicitly set the store? I thought the Ctx or Apparatus would do that.

File: docs/gpt-sweep/report.py [hunk 4]
```diff
@@ -5,23 +5,39 @@
...
-        _here = Path('docs/gpt-sweep')
-    RESULTS_PATH = _here / 'results.json'
+    # The experiment publishes its curves to the project-scoped store under this
+    # name (see experiment.py); we resolve them by name at export time, so no data
+    # file is committed to Git. The store is the HF bucket when configured, else local.
+    CURVES_REF = 'reports/gpt-sweep/curves'
+
+    def load_curves() -> dict[str, list[float]]:
+        """Resolve `{arch|lr: [val_loss per epoch]}` from the store, or `{}` if unpublished."""
+        store = LocalApparatus('gpt-sweep').store()
```
Comment (z0u) [inline docs/gpt-sweep/report.py]: Could we use default_store here? Or would that not work out to be simpler/more decoupled?

File: src/experiment/model/README.md [hunk 5]
```diff
@@ -98,7 +98,7 @@ $$h \leftarrow \text{Norm}\big(h + \alpha\,\text{sub}(h)\big)$$
...
+The full nGPT recipe goes one step further: it takes the step _towards_ the sub-module's normalized output, $h \leftarrow \text{Norm}\big(h + \alpha\,(h_\text{sub} - h)\big)$, with $\alpha$ a learned per-**channel** vector (the paper's _eigen learning rates_). We deliberately strip that back to a scalar. The [notebook sweep](../../../docs/gpt-sweep/report.py) shows the simplification is free at this scale: the scalar gate matches per-channel nGPT, while dropping the gate entirely (a raw additive retraction) breaks training outright. The learned gates also settle near $1/n_\text{layer}$ on average — the value that keeps the twelve-layer stack moving an $O(1)$ distance overall.
```
Comment (z0u) [inline src/experiment/model/README.md:101]: I think we can use repo-based absolute paths like `/docs`. Works in VS Code, unsure about other renderers.

File: src/mini/local_apparatus.py [hunk 6]
```diff
@@ -105,7 +105,11 @@ async def amap(
         sizes = [len(it) for it in iterables_lists]
         n = min(sizes) if sizes else None
 
-        log.info('Running %d jobs with %d workers', n, self.max_workers)
+        # Name the backend (symmetric with Modal's 'Running … on Modal'), so a
+        # local run — e.g. a fallback when a notebook meant to use Modal — is
+        # visible in the logs rather than only inferable from the *absence* of
+        # Modal's image-build output.
+        log.info('Running %d jobs locally on CPU (%d workers)', n, self.max_workers)
```
Comment (z0u) [inline src/mini/local_apparatus.py]: Good, but there may be a local GPU?
